proxy/src/lib.rs

```diff
@@ -366,7 +366,12 @@ where
                     http::StatusCode::INTERNAL_SERVER_ERROR
                 }
             }
-        })
+        });
+
+        // Install the request open timestamp module at the very top
+        // of the stack, in order to take the timestamp as close as
+        // possible to the beginning of the request's lifetime.
+        telemetry::sensor::http::TimestampRequestOpen::new(map_err)
```

Comment:
let's put this around `router` instead of around `map_err`, so that we still uniformly handle errors. Basically, i think that MapErr should wrap _everything_ unless there's a good reason not to.

Reply:
Okay, sounds good.